lib/terraforming/resource/s3.rb

```diff
@@ -53,7 +54,9 @@ def bucket_policy_of(bucket)
       end
 
       def buckets
-        @client.list_buckets.map(&:buckets).flatten.select { |bucket| same_region?(bucket) }
+        @client.list_buckets.map(&:buckets).flatten.select do |resource|
+          @match_regex ? module_name_of(resource) =~ @match_regex : 1
+        end.select { |bucket| same_region?(bucket) }
```

Comment:
__[rubocop]__
* Avoid multi-line chains of blocks. [:ref](https://github.com/bbatsov/ruby-style-guide#single-line-blocks)

<!-- Comment by [SideCI](https://sideci.com) -->